### PR TITLE
[FLINK-5386][Table API & SQL] refactoring Window Clause

### PR DESCRIPTION
**Pull request by sunjincheng121**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [×] General
  - The pull request references the related JIRA issue ("[FLINK-5386][Table API & SQL] refactoring Window Clause")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [ ] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [×] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/table.scala [hunk 1]
```diff
@@ -785,13 +793,19 @@ class Table(
...
+  def window(groupWindow: GroupWindow): Table = {
     if (tableEnv.isInstanceOf[BatchTableEnvironment]) {
       throw new ValidationException(s"Windows on batch tables are currently not supported.")
     }
-    new GroupWindowedTable(this, Seq(), groupWindow)
+    if (None == groupWindow.name) {
+      throw new ValidationException("An alias must be specified for the window.")
+    }
+    if (windowPool.contains(groupWindow.name.get)) {
+      throw new ValidationException("The window alias can not be duplicated.")
```
Comment (shaoxuan-wang) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/table.scala]: s/"The window alias can not be duplicated."/"The same window alias has been defined"/

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/windows.scala [hunk 2]
```diff
@@ -150,7 +147,7 @@ class TumblingWindow(size: Expression) extends GroupWindow {
   def as(alias: String): TumblingWindow = as(ExpressionParser.parseExpression(alias))
 
   override private[flink] def toLogicalWindow: LogicalWindow =
-    ProcessingTimeTumblingGroupWindow(alias, size)
+    ProcessingTimeTumblingGroupWindow(name, size)
```
Comment (shaoxuan-wang) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/windows.scala]: Better to keep using "alias" here. By the way,
I found that the input of some functions use "name", like ProcessingTimeTumblingGroupWindow, while some others use "alias", like TumblingEventTimeWindow. Better to change them consistently to "alias". What do you think.

Comment (sunjincheng121) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/windows.scala]: Hi, @shaoxuan-wang  thanks a lot for the review. I have updated the PR according to your comments. The change list:
 1. Remove GroupWindowedTable.
 2. Chanage "name" to "alias".
 3. Add testMultiWindow.
 4. Rebase code and fixed the conflicts.

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/dataset/DataSetWindowAggregateITCase.scala [hunk 3]
```diff
@@ -116,4 +119,21 @@ class DataSetWindowAggregateITCase(configMode: TableConfigMode)
     val results = windowedTable.toDataSet[Row].collect()
     TestBaseUtils.compareResultAsText(results.asJava, expected)
   }
+
+  @Test(expected = classOf[ValidationException])
+  def testMutilGroupWindow(): Unit = {
```
Comment (fhueske) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/dataset/DataSetWindowAggregateITCase.scala]: Mutil -> Multi

**Comment by sunjincheng121** [comment]
Hi, @fhueske thanks a lot for the review. I have updated the PR according to your comments. Let me know if I miss something，thanks again!!

**Comment by fhueske** [comment]
Thanks for the update @sunjincheng121!
I'll do one more pass over the docs and merge this PR.

**Pull request closed without merging**